python-avd/pyavd/_eos_designs/shared_utils/wan.py

```diff
@@ -83,25 +83,68 @@ def wan_interfaces(self: SharedUtils) -> EosDesigns._DynamicKeys.DynamicNodeType
             [interface for interface in self.l3_interfaces if interface.wan_carrier]
         )
+        # TODO: We could have a scenario where there may be no L3 interfaces configured and one or more
+        # interfaces under L3 Port-Channels might be configured as wan-interfaces.
+        # In such cases, below error may not be accurate.
+        # Also need to avoid potential recursion
+        # i.e. avoid logic in self.wan_interfaces referencing self.wan_port_channels and vice-versa
         if not wan_interfaces:
             msg = "At least one WAN interface must be configured on a WAN router. Add WAN interfaces under `l3_interfaces` node setting with `wan_carrier` set."
             raise AristaAvdError(msg)
         return wan_interfaces
 
+    @cached_property
+    def wan_port_channels(self: SharedUtils) -> EosDesigns._DynamicKeys.DynamicNodeTypesItem.NodeTypes.NodesItem.L3PortChannels:
+        """Interfaces under node config l3_port_channels can be considered as WAN-facing port-channel interfaces."""
+        if not self.is_wan_router:
+            return EosDesigns._DynamicKeys.DynamicNodeTypesItem.NodeTypes.NodesItem.L3PortChannels()
+
+        # TODO: revise validation logic
+        # Ideally we need to raise exception when we cannot find a single interface within `L3 Interfaces` and `L3 Port-Channels` with `wan_carrier` set.
+        # Could we validate this condition outside of logic for each of cached_property?
+        return EosDesigns._DynamicKeys.DynamicNodeTypesItem.NodeTypes.NodesItem.L3PortChannels(
+            [port_channel for port_channel in self.l3_port_channels if port_channel.wan_carrier]
+        )
+
     @cached_property
     def wan_local_carriers(self: SharedUtils) -> list:
         """
-        List of carriers present on this router based on the wan_interfaces with the associated WAN interfaces.
+        List of carriers present on this router based on the wan_interfaces and wan_port_channels with the associated WAN interfaces.
 
             interfaces:
               - name: ...
                 ip: ... (for route-servers the IP may come from wan_route_servers).
         """
         if not self.is_wan_router:
             return []
-
-        local_carriers_dict = {}
-        for interface in self.wan_interfaces:
+        # We would like to combine info from both L3 Interfaces and L3 Port-Channels configured as wan interfaces
+        # and represent this for each carrier
+        carriers_dict = {}
+        carriers_dict = self.get_wan_local_carriers(carriers_dict, self.wan_interfaces)
+        # modify dictionary from above step with any L3 port-channel based wan interfaces
+        # associated with carrier
+        carriers_dict = self.get_wan_local_carriers(carriers_dict, self.wan_port_channels)
+        return list(carriers_dict.values())
+
+    def get_wan_local_carriers(
+        self: SharedUtils,
+        local_carriers_dict: dict | None,
```

Comment:
I don't think we need the None variant here, since you always get a dict. So you can also remove the extra logic for the none case.

Reply:
Done